pkg/extkingpin/path_content_reloader.go

```diff
@@ -22,74 +20,65 @@ type fileContent interface {
 	Path() string
 }
 
-// PathContentReloader starts a file watcher that monitors the file indicated by fileContent.Path() and runs
-// reloadFunc whenever a change is detected.
-// A debounce timer can be configured via opts to handle situations where many "write" events are received together or
-// a "create" event is followed up by a "write" event, for example. Files will be effectively reloaded at the latest
-// after 2 times the debounce timer. By default the debouncer timer is 1 second.
-// To ensure renames and deletes are properly handled, the file watcher is put at the file's parent folder. See
-// https://github.com/fsnotify/fsnotify/issues/214 for more details.
+// PathContentReloader runs the reloadFunc when it detects that the contents of fileContent have changed.
 func PathContentReloader(ctx context.Context, fileContent fileContent, logger log.Logger, reloadFunc func(), debounceTime time.Duration) error {
 	filePath, err := filepath.Abs(fileContent.Path())
 	if err != nil {
 		return errors.Wrap(err, "getting absolute file path")
 	}
-
-	watcher, err := fsnotify.NewWatcher()
 	if filePath == "" {
 		level.Debug(logger).Log("msg", "no path detected for config reload")
 	}
-	if err != nil {
-		return errors.Wrap(err, "creating file watcher")
+
+	engine := &pollingEngine{
+		filePath:   filePath,
+		logger:     logger,
+		debounce:   debounceTime,
+		reloadFunc: reloadFunc,
 	}
-	go func() {
-		var reloadTimer *time.Timer
-		if debounceTime != 0 {
-			reloadTimer = time.AfterFunc(debounceTime, func() {
-				reloadFunc()
-				level.Debug(logger).Log("msg", "configuration reloaded after debouncing")
-			})
-			reloadTimer.Stop()
+	return engine.Start(ctx)
+}
+
+// pollingEngine is an implementation of reloaderEngine that keeps rereading the contents at filePath and when the
+// checksum changes it runs the reloadFunc.
```

Comment:
No interface needed now 🙂 
```suggestion
// pollingEngine keeps rereading the contents at filePath and runs 
// reloadFunc when the checksum changes.
```